### PR TITLE
fix(RHINENG-4565): Refactor staleness GET endpoint and add permission tests

### PR DESCRIPTION
**Pull request by fellipeh**
# Overview

This PR is being created to address [RHINENG-4565](https://issues.redhat.com/browse/RHINENG-4565).

- Make sure we test RBAC permissions to GET staleness endpoint
- Refactor staleness GET endpoint

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by kruai — CHANGES_REQUESTED** [review]
Looks good, though I think it needs a test to make sure that verifies that the endpoint works correctly when an `rbac_filter` is provided (i.e. when the RBAC response has an `attributeFilter`, which is what was breaking it originally)

File: tests/test_api_staleness_get.py [hunk 1]
```diff
@@ -55,3 +57,18 @@ def test_get_sys_default_staleness(api_get):
     assert response_data["immutable_time_to_stale"] == expected_result["immutable_time_to_stale"]
     assert response_data["immutable_time_to_stale_warning"] == expected_result["immutable_time_to_stale_warning"]
     assert response_data["immutable_time_to_delete"] == expected_result["immutable_time_to_delete"]
+
+
+def test_get_staleness_rbac_denied(subtests, mocker, api_get, db_get_staleness_culling, enable_rbac):
+    get_rbac_permissions_mock = mocker.patch("lib.middleware.get_rbac_permissions")
+    url = build_sys_default_staleness_url()
+
+    for response_file in STALENESS_READ_PROHIBITED_RBAC_RESPONSE_FILES:
```
Comment (thearifismail) [inline tests/test_api_staleness_get.py:66]: Should there be a test with some valid RBAC permissions?

**Review by thearifismail — COMMENTED** [review]
Looks good with just one question?

File: tests/test_api_staleness_get.py (hunk 1 above)
Comment (fellipeh) [reply tests/test_api_staleness_get.py:66]: done

**Review by thearifismail — APPROVED** [review]
LGTM

File: tests/test_api_staleness_get.py [hunk 2]
```diff
@@ -24,9 +29,10 @@ def test_get_default_staleness(api_get):
     assert response_data["immutable_time_to_delete"] == expected_result["immutable_time_to_delete"]
 
 
-def test_get_custom_staleness(api_create_staleness, api_get):
+@pytest.mark.parametrize("rbac_filter", ["test"])
+def test_get_custom_staleness(api_create_staleness, api_get, rbac_filter):
     created_response_status, _ = api_create_staleness(_INPUT_DATA)
-    url = build_staleness_url()
+    url = build_staleness_url(query=f"?rbac_filter={rbac_filter}")
```
Comment (kruai) [inline tests/test_api_staleness_get.py:35]: `rbac_filter` isn't a query param and doesn't make sense in this context. In order to replicate the scenario this PR is fixing, you need to set the attributeFilter of the RBAC response. Here's an example where we do this: https://github.com/RedHatInsights/insights-host-inventory/blob/master/tests/test_xjoin.py#L1452

**Review by kruai — CHANGES_REQUESTED** [review]

File: tests/test_api_staleness_get.py [hunk 3]
```diff
@@ -24,9 +29,9 @@ def test_get_default_staleness(api_get):
     assert response_data["immutable_time_to_delete"] == expected_result["immutable_time_to_delete"]
 
 
-def test_get_custom_staleness(api_create_staleness, api_get):
-    created_response_status, _ = api_create_staleness(_INPUT_DATA)
-    url = build_staleness_url()
+@pytest.mark.parametrize("rbac_filter", ["test"])
+def test_get_custom_staleness(api_create_staleness, api_get, rbac_filter):
+    url = build_sys_default_staleness_url()
```
Comment (kruai) [inline tests/test_api_staleness_get.py:34]: The `rbac_filter` param is no longer used, so this change should probably just be reverted

File: tests/test_api_staleness_get.py [hunk 4]
```diff
@@ -55,3 +60,35 @@ def test_get_sys_default_staleness(api_get):
     assert response_data["immutable_time_to_stale"] == expected_result["immutable_time_to_stale"]
     assert response_data["immutable_time_to_stale_warning"] == expected_result["immutable_time_to_stale_warning"]
     assert response_data["immutable_time_to_delete"] == expected_result["immutable_time_to_delete"]
+
+
+def test_get_staleness_rbac_denied(subtests, mocker, api_get, db_get_staleness_culling, enable_rbac):
```
Comment (kruai) [inline tests/test_api_staleness_get.py:65]: It doesn't look like `db_get_staleness_culling` is used anywhere

File: tests/test_api_staleness_get.py [hunk 5]
```diff
@@ -55,3 +60,35 @@ def test_get_sys_default_staleness(api_get):
...
+
+        with subtests.test():
+            get_rbac_permissions_mock.return_value = mock_rbac_response
+
+            response_status, _ = api_get(url)
+
+            assert_response_status(response_status, 403)
+
+
+def test_get_staleness_rbac_allowed(subtests, mocker, api_get, db_get_staleness_culling, enable_rbac):
```
Comment (kruai) [inline tests/test_api_staleness_get.py:82]: It doesn't look like `db_get_staleness_culling` is used anywhere here either

**Review by kruai — CHANGES_REQUESTED** [review]

**Review by kruai — APPROVED** [review]

**Pull request merged**